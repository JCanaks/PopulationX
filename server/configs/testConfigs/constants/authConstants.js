export const SIGNUP_USER = `
    mutation signup($name: String!, $email: String!, $password: String!) {
      signup(name: $name, email: $email, password: $password) {
        user {
          name
          email
        }
      }
    }
`;

export const LOGIN_USER = `
    mutation login($email: String!, $password: String!) {
      login(email: $email, password: $password) {
        user {
          name
          email
        }
      }
    }
`;

export const GET_WELCOME_MESSAGE = `
query {
  info
}`;
