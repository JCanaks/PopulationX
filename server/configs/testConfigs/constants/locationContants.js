export const CREATE_LOCATION = `
    mutation createLocation($name: String!, $noOfFemales: Int!, $noOfMales: Int!, $locatedIn: String) {
      createLocation(name: $name, noOfFemales: $noOfFemales, noOfMales: $noOfMales, locatedIn: $locatedIn) {
        name
        noOfFemales
        noOfMales
        totalResidents
        parentLocation {
          name
        }
        childLocations {
          name
        }
      }
    }
`;

export const UPDATE_LOCATION = `
    mutation updateLocation($id: ID!, $name: String, $noOfFemales: Int, $noOfMales: Int, $locatedIn: String) {
      updateLocation(id: $id, name: $name, noOfFemales: $noOfFemales, noOfMales: $noOfMales, locatedIn: $locatedIn) {
        name
        noOfFemales
        noOfMales
        totalResidents
        parentLocation {
          name
        }
        childLocations {
          name
        }
      }
    }
`;

export const GET_LOCATIONS = `
    query {
        allLocations {
        name
        noOfFemales
        noOfMales
        totalResidents
        parentLocation {
          name
        }
        childLocations {
          name
        }
      }
    }
`;

export const DELETE_LOCATION = `
    mutation deleteLocation($id: ID!) {
        deleteLocation(id: $id) {
        name
      }
    }
`;
