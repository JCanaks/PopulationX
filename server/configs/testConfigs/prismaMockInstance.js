import { makePrismaClientClass } from 'prisma-client-lib';
import { typeDefs } from '../../generated/prisma-client/prisma-schema';
import { models } from '../../generated/prisma-client';

const PrismaMockInstance = makePrismaClientClass({
  typeDefs,
  models,
  endpoint: 'http://localhost:4499',
});
export default new PrismaMockInstance();
