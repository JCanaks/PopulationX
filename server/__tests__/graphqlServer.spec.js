import '@babel/polyfill';
import prismaMockInstance from '../configs/testConfigs/prismaMockInstance';
import createGraphqlTestServer from '../configs/testConfigs/testClient';
import { GET_WELCOME_MESSAGE } from '../configs/testConfigs/constants/authConstants';

const { query } = createGraphqlTestServer();

describe('Testing default home resolver in the graphQL server', () => {
    test('Introductory default welcome message', async () => {
      const { errors, data } = await query({
        query: GET_WELCOME_MESSAGE,
        variables: {},
      });
  
      expect(data).toBeDefined();
      expect(data).toMatchSnapshot();
      expect(errors).toBeUndefined();
    });
  });
  