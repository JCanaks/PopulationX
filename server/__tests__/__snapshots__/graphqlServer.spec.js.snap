// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing default home resolver in the graphQL server Introductory default welcome message 1`] = `
Object {
  "info": "Welcome to PopulationX, an Easy-to-use Population Management System",
}
`;
